src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,70 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
 public class WorkWithFile {
+    public static final String SUPPLY_FIELD = "supply";
+    public static final String BUY_FIELD = "buy";
+    public static final String RESULT_FIELD = "result";
+    public static final int NAME_INDEX = 0;
+    public static final int VALUE_INDEX = 1;
+    public static final String DELIMITER = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        Map<String, Integer> data = readDataFromFile(fromFileName);
+        String report = createReport(data);
+        writeReportToFile(report, toFileName);
+    }
+
+    private void writeReportToFile(String report, String toFileName) {
+        File file = new File(toFileName);
+        try {
```

Comment:
[Use try-with-resourses here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#try-with-resources)
Here is enough to make something like 
```
try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(fileName))) {
         bufferedWriter.write(report);
```